File: CompanyApi/src/CompanyApi/CompanyApi/Models/Company.cs
using System.ComponentModel.DataAnnotations;

namespace CompanyApi.Models;

public class Company
{
    public int Id { get; set; }

    [Required]
    [MaxLength(255)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    public string StockTicker { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string Exchange { get; set; } = string.Empty;

    [Required]
    [MaxLength(12)]
    public string Isin { get; set; } = string.Empty;

    [MaxLength(255)]
    public string? Website { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
} 

File: company-client/src/types/company.ts
export interface Company {
    id: number;
    name: string;
    stockTicker: string;
    exchange: string;
    isin: string;
    website?: string;
    createdAt: string;
    updatedAt: string;
}

export interface CreateCompanyDto {
    name: string;
    stockTicker: string;
    exchange: string;
    isin: string;
    website?: string;
}

export interface UpdateCompanyDto extends CreateCompanyDto {
    id: number;
} 